.github/workflows/ci.yml

```diff
@@ -9,30 +9,54 @@ on:
 
 jobs:
 
+  check-systems:
+    name: Check runix ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@v22
+        with:
+          extra_nix_config: |
+            experimental-features = nix-command flakes
+            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+            substituters = https://cache.nixos.org
+            trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+            max-jobs = auto
+            cores = 0
+            substitute = true
+            builders-use-substitutes = true
+            fallback = true
+            connect-timeout = 5
+            stalled-download-timeout = 90
+            timeout = 0
+
+      - name: check clippy
+        run: nix develop '.#rust-env' -c cargo clippy
+
+      - name: check format
+        run: nix develop '.#rust-env' -c cargo fmt --check
+
+      - name: run tests
+        run: nix develop '.#rust-env' -c cargo test
+
   check:
-    name: Check runix
+    if: ${{ always() }}
```

Comment:
What's this for?

Reply:
This aligns with #34.

There's an issue with `flox develop <TARGET> --command <CMD>;` that prevents us from using `flox` here.

As an added bonus this breaks the dependency cycle that currently exists between `flox` and `runix` which has caused several issues with lockfile inheritance in the past. Since `flox` depends on `runix`, `runix` shouldn't depend on `flox` to avoid a bootstrap paradox.